app-website-4/ProductsApi/Auth/TokenProvider.cs

```diff
@@ -0,0 +1,32 @@
+﻿using Microsoft.IdentityModel.Tokens;
+using System;
+using System.IdentityModel.Tokens.Jwt;
+using System.Security.Claims;
+using System.Text;
+
+namespace ProductsApi.Auth
+{
+    public class TokenProvider
+    {
+        public string GenerateToken(string[] usernameAndPassword)
+        {
+            if (usernameAndPassword[0] == "admin" && usernameAndPassword[1] == "password")
+            {
+                var claimsdata = new[] { new Claim(ClaimTypes.Name, usernameAndPassword[0]) };
+                var key = new SymmetricSecurityKey(Encoding.UTF8.GetBytes("This is a secret key"));
+                var signInCred = new SigningCredentials(key, SecurityAlgorithms.HmacSha256Signature);
+                var token = new JwtSecurityToken(
+                     issuer: "Products API",
+                     audience: "Products App",
+                     expires: DateTime.Now.AddMinutes(30),
```

Comment:
👍 Short-lived tokens are always preferable over long-lived. Especially if you don't have a mechanism for revoking these tokens - which you don't in this kata.

Is there something on the frontend to refresh them?

Reply:
No, not at the moment. Could you point me in the direction of how we do this?

Reply:
It's a much bigger problem when it comes to JWT's.

One of the nice things is JWT's allows your API's to be stateless. What I mean by that is they can validate someone is authenticated/authorized based _purely_ on the data they send to you - the token. No need to check a database.

However... if you wanted to revoke a token, suddenly you need to do some bookkeeping yourself. This might be a fast Redis-like cache which maps a token to a true/false flag indicating whether it's been revoked. That means every time you receive a token, you need to check it against the cache/database.

So, if you wanted to do revoking of tokens then you've got a lot more work ahead of you.

This is a good read: https://stackoverflow.com/a/40385939/5235388